e2e/testcafe-devextreme/tests/accessibility/common/dateBox/buttonsAndLabels.ts

```diff
@@ -0,0 +1,71 @@
+import { ClientFunction } from 'testcafe';
+import { Properties } from 'devextreme/ui/date_box.d';
+import DateBox from 'devextreme-testcafe-models/dateBox';
+import url from '../../../../helpers/getPageUrl';
+import { defaultSelector, testAccessibility, Configuration } from '../../../../helpers/accessibility/test';
+import { Options } from '../../../../helpers/generateOptionMatrix';
+
+fixture.disablePageReloads`Accessibility`
+  .page(url(__dirname, '../../../container.html'));
+
+const now = new Date();
+
+const options: Options<Properties> = {
+  value: [now],
+  label: [undefined, 'label'],
+  showClearButton: [true, false],
+  showDropDownButton: [true, false],
+  inputAttr: [{ 'aria-label': 'aria-label' }],
+  buttons: [
+    undefined,
+    [
+      {
+        name: 'today',
+        location: 'before',
+        options: {
+          text: 'Today',
+          stylingMode: 'text',
+          onClick: () => {},
+        },
+      },
+    ],
+  ],
+};
+
+const created = async (t: TestController, optionConfiguration): Promise<void> => {
+  const {
+    disabled, readOnly, type, value,
+  } = optionConfiguration;
+
+  if (
+    disabled
+    || readOnly
+    || (type === 'datetime' && !value)
+  ) {
+    return;
+  }
```

Comment:
looks like this condition is always false, no?

Reply:
https://github.com/DevExpress/DevExtreme/pull/29875/commits/214753b832bfe3af6e6c2d8497e50e05cefa5484